x-pack/plugins/reporting/server/lib/export_types_registry.ts

```diff
@@ -88,13 +69,14 @@ export class ExportTypesRegistry {
   }
 }
 
+// TODO: define a 2nd export type registry for "immediate execute" report job types.
+// That type does not need a `CreateJobFn` in the definition. Then remove the below `any`s
```

Comment:
@joelgriffith I'm curious what you think about this.

I hope something like this would allow us to have more harmony with immediate / async types of exports

Reply:
I don't think we necessarily should be spending more of our time on the immediate export in favor of doing it client-side. The future lies in doing these types of the things purely browser-side, for smaller result sets, and just doing "raw" exports of ES data reporting-side...

Reply:
Anything that can get us favor now is good, but don't want to spend too many cycles on it :)

Reply:
Rewrote the comment: 945ace999a4